### ISSUE TITLE
Update fretboard diagram when its attached chord symbol is changed

### ISSUE THREAD

**Issue by avvvvve** [musescore/MuseScore#25462]
### Problem

When you add a fretboard diagram to a score, it comes with an attached chord symbol element. Double-clicking into the chord symbol text lets you edit the chord (which is reflected correctly in playback), but doing so does not update the attached fretboard diagram.

https://github.com/user-attachments/assets/99d30673-0bb1-40bf-8342-19f24c0a7777

### Solution

Fretboard diagrams need to check what the chord symbol they are attached to is, and update accordingly.

For example, this should be possible:

1. Open the 'Fretboard diagram' palette and add a C chord.
2. Double click on the 'C' chord symbol in the score
3. Type in 'Cm' and click out of the field
4. See that the fretboard diagram updates to the Cm shape

This means that each chord symbol that MuseScore Studio can play back also needs a fretboard diagram assigned to it, which is a larger task to be completed alongside this one, likely with @bkunda's help. (Eventually we will add options to choose between different diagram shapes/positions for the same chord)

This is also the first step to enabling [transposition of fretboard diagrams and chord symbols](https://github.com/musescore/MuseScore/issues/23712).

**Comment by scorster** [comment]
@ [avvvvve](https://github.com/avvvvve) **Paraphrased** *Fretboard diagrams should check the chord quality of the chord symbol they to which they are attached, and update accordingly when the user changes the chord symbol.*

Nice idea!

And if there are multiple fretboard diagrams for the new selected chord symbol, like the exampled Cm, then show some sort of dialog or picker where the user can easily select among the C minor fretboard diagrams. And I can see the helpfulness of the option to choose "No fretboard diagram."

**Comment by bkunda** [comment]
> like your exampled Cm, then show some sort of dialog or picker where the user can easily select among C minor fretboard diagrams voicings

We will eventually do this, but it's out of scope for the initial feature release.

**Comment by scorster** [comment]
I'd also like to see:

• Fretboard Diagrams realized from a chord
• Fretboard Diagrams realized from a bass note and a subsequent chord (as we hear in styles like bossa nova.
• Fretboard Diagrams realized from a scale (or just a sequence of notes)

And the inverse:

• Notation generated from Freboard Diagrams

**Comment by avvvvve** [comment]
@scorster

> Fretboard Diagrams realized from a chord...

Please create a feature request (or requests) for those three bullets as it's a separate ask from the issue here. Like Bradley said, we'd like that eventually too, but it's more likely to happen sooner if a community member picks that up as a project.

> Notation generated from Fretboard Diagrams

Once this issue is resolved, you'll be able to enter a chord symbol, have an appropriate fretboard diagram show up automatically, and then that chord can be realized in notation using the existing "Realize chord symbols" function (right click on a chord symbol to see this).

**Comment by scorster** [comment]
> @scorster
> 
> > Fretboard Diagrams realized from a chord...
> 
> Please create a feature request (or requests) for those three bullets as it's a separate ask from the issue here. Like Bradley said, we'd like that eventually too, but it's more likely to happen sooner if a community member picks that up as a project.

Will do. And thanks again for your interest and work on this.

> > Notation generated from Fretboard Diagrams
> 
> Once this issue is resolved, you'll be able to enter a chord symbol, have an appropriate fretboard diagram show up automatically, and then that chord can be realized in notation using the existing "Realize chord symbols" function (right click on a chord symbol to see this).

If I'm following correctly:

• If there's a chord symbol and no fretboard diagram, the user right-clicks on the chord symbol, then chooses Realize Chord Symbols, and  MuseScore will pick a "canned" or use the selected "override" voicing.

• If there's a chord symbol AND a fretboard diagram the user can right-click the chord symbol, choose Realize Chord Symbols, and MuseScore will realize notation *based on the fretboard diagram*?

I think that would be okay, but maybe the following approach would be clearer and more versatile 

When the user right-clicks the fretboard they get a contextual menu with a **Realize Fretboard Diagram.** That way, via the contextual menu, they can generate notation from the chord symbol OR the from fretboard diagram.

**Comment by avvvvve** [comment]
Yes, eventually it would be great to have that level of customization over realizing chords, but the ability to read the notes indicated by a fretboard diagram is out of scope for this particular task.
